### ISSUE TITLE
Update 2018-03-31 ~ gbXML Viewer R12 ~ Stable Release Candidate

### ISSUE THREAD

**Issue by theo-armour** [ladybug-tools/spider#99]
@ladybug-tools/spiders 

![image](https://user-images.githubusercontent.com/547626/38168545-2cb8c140-3504-11e8-9e8c-41f4ce5d2b8c.png)

_Note the shadows created via the newly added Sun Range modules as well as exterior walls colored according to the eight compass directions they face._

## [gbXML Viewer R12.13 ]( http://www.ladybug.tools/spider/gbxml-viewer/dev )

Many new features. The new numbers menu provide many new stats for areas, window wall ratios and more. User selected user interface colors is a new feature nobody asked for but it was just fun to do. ;-)

Fingers crossed this release become the stable release very soon. We say this because R13 is coming along really nicely.

## Change Log

R12.13
APP/ All modules
* Fix issues with menu buttons not resetting style on new model load

R12.12
CSS / APP
* Add user selected color themes

APP
* Turn off automatic load of saved changes

R12.10
COR / Core Modules
* Simplify slider code
	* sticks to side
	* Should work better on a phone
* Update core.html so shows all three windows
	* Follows APP menu more closely
CSS
* Works with new slider code
* Various ID name changes

R12.9
CSS
* Widths to rems
* Most all colors by css vars

APP / Application Module
* Rejigged menu / item positions

NUM / Numbers Module
* 2018-03-23 ~ Standardize width of buttons
* Add Michalina's new orientation color palette
* Menu uses HTML details
* Menu add toggle visability buttons
* Add more Orientation features WWR etc

R12.8
* NUM / Numbers Module
	* 2018-03-22 ~ Add select by storey / show in 3D / display attributes
	* 2018-03-22 ~ Add Zones
	* 2018-03-22 ~ Button to show surfaces for total floor area
	* 2018-03-22 ~ Button to show surfaces for exterior area
	* 2018-03-22 ~ Add show orientation by color
	* 2018-03-22 ~ Add show openings / make clickable / App to report
* APP / Application Module
	* Fix reports menu not deleting itself
	*  Make sample file more visible
* HUD / Heads-up Display
	* Fix spaces not showing in HUD select space

Some issues with update in the HUD.

R12.7
* NUM/ Numbers Module
	* Add many new numbers

R12.6
* APP/ Style
	* Trying to learn how to control right menu / let it flow and grow nicely
	* Make 3D model more visible on loading
* APP/Menu
	* Only load changes JSON file when initial default model is loaded
* ISS/Issues module
	* Add 'Surfaces wth close vertex' item
* NUM/Numbers module
	* First commit
R12.5
* SAV/ Save changes
	* Add load ground plane
	* Add load background gradient
	* Add ability to parse files at load time

R12.4
* ISS/Issues module:
	* Add check for undefined CAD Object ID
	* Add check for tiny surfaces
	* Add check for surfaces that contain other surfaces or overlap
		* Currently turned off because uses much time and has found none so far
		* Work-in-progress

R12.3 ~ Most modules working
* Reports 2 menu: simpler code / faster / workflow more streamlined
* Issues: split off from reports / Adjacent spaces items now sorted
* Save changes: now supports loading color schemes
* Right menu has slider

Goals for R12

* Simplify the core modules quite a bit
	* Make things so that Python coders say it's almost as easy as Python
* Split Reports into two modules
	* Reports - Identify spaces, surfaces, storeys etc
		Make selecting things as easy as Heads-up display
	* Issues - Identify duplicate surfaces or surfaces with two identical adjacent spaces etc / find the problem areas
* Maybe start an analysis effort that includes things like Window Wall Ratios

APP and all the test files now seem to co-exist fairly well
* This will be ongoing effort
* Every module in its own folder
* Index and test HTML will be identical in all folders
* Only .js  and read me files in each module need updating

R12 First Commit
* Big effort to simplify the core scripts and streamline the loading process
* The index.html for each module folder automatically load the read me file whatever folder its in
	* You can move or copy this index.html file to any folder and it just works
* Standard gv-tmp.hml file to test JavaScript modules. Almost drag and drop into any folder
* All modules use same style sheet
	* Closes: 2018-03-04 ~ Use main style sheet

**Comment by mdengusiak** [comment]
@theo-armour I am doing now final tests on this version, so we can publish this soon and make R12  stable version

**Comment by mdengusiak** [comment]
@theo-armour 
I tested now all clicks and full functionality and here is my summary comments
1. I am experiencing problem with `Issues`
when I am keep pressing Issue I am experiencing this duplicates and can not see right information.
![image](https://user-images.githubusercontent.com/14005893/38217648-0738899a-36c7-11e8-88f6-c495a8d5fe17.png)

2. Unable to switch off `Openings` even after `reset views`
![image](https://user-images.githubusercontent.com/14005893/38217742-516aafd4-36c7-11e8-95c3-0b629f7dd170.png)

3. Issue with disapearing surface
![image](https://user-images.githubusercontent.com/14005893/38217969-21d2bfa4-36c8-11e8-92d9-3de8f4a9255a.png)

4 . `Surface by Type `does not work after Windows and on clean file as well
![image](https://user-images.githubusercontent.com/14005893/38218163-c2b204e8-36c8-11e8-9e74-0da6289cce35.png)

![image](https://user-images.githubusercontent.com/14005893/38218299-2c966c5a-36c9-11e8-84c8-1be785f110fd.png)

5. Numbers
`Area by Surface Type` is not isolating elements in 3D
![image](https://user-images.githubusercontent.com/14005893/38218381-804cea86-36c9-11e8-9cbf-a6a0b8bde273.png)

6. Right Menu
those two are not responding for me 
![image](https://user-images.githubusercontent.com/14005893/38218532-028a6a64-36ca-11e8-95cb-11ef75c7187d.png)

**Comment by theo-armour** [comment]
@mdengusiak 

Tank you very much for this excellent quality assurance testing.

My responses are as follows:

> 1 I am experiencing problem with Issues

Ditto with Reports. Fix in R12.15


> 2. Unable to switch off Openings even after reset views

I'm having trouble reproducing in R12.15


> 3. Issue with disappearing surface

If this occurs when there are duplicate surface and/or duplicate adjacent spaces, then I know about this issue

Added this text to issues: 'When there are duplicate surface and/or duplicate adjacent spaces, toggling the visibility of one of the items may cause both items not to be visible.'


> 4. Surface by Type does not work after Windows and on clean file as well

Of course it did not work - because I had moved the function instead of copying the function to another file.. ;-)

Fix in R12.15


> 5. Numbers Area by Surface Type is not isolating elements in 3D

Same as issue 4

> Right Menu those two are not responding for me

You may have to click the 'update' button in order to force an update. 

Reason: there are so many things going on in left menu, in-world, input box and select box that keeping track of all the possible events is still a work-in-progress. Fingers-crossed R13 gets better to simplifying the interaction.

**Comment by mdengusiak** [comment]
@theo-armour 
we almost there :)
1. Fixed
2. When I click Numbers and then Reset view I still see Black openings, aslo Toggle Openings does not work for me. If we have this working if version R13 we can skip this but we can not switch off opening right now. 
3. OK
4. Fixed
5.  Fixed

6. there is one more new item
when we have `Air` Type and want to change to Internal wall, or Floor this is fine but when I try to assign CADObjectID we got this message. I think we got this working before.  
![image](https://user-images.githubusercontent.com/14005893/38392554-d3c7f126-391f-11e8-96b9-6d1d43834caf.png)

**Comment by theo-armour** [comment]
@mdengusiak 

> #6  I think we got this working before.

I have looked through a number of previous revisions and cannot find where we fixed this. 

What I do remember is that we went from not being able to edit any CADObjectID  to being able to edit them.

I any case, what we want for R13 is being able to edit the group type and not the particular individual ID.

So maybe we should let this go for the time being?
